assets/js/app/pgpparsing.js

```diff
@@ -0,0 +1,55 @@
+/* 
+ * Copyright (C) 2018 crashdemons (crashenator at gmail.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+if (typeof Lineparser === "undefined") {
+    var Lineparser = {};//suppress warnings in editor - doesn't change anything.
+    console.error("pgpparser code loaded before Lineparser");
+}
+
+if (typeof PgpClearsignParser === "undefined") {
+    var PgpClearsignParser = {};//suppress warnings in editor - doesn't change anything.
+    console.error("pgpparser code loaded before PgpClearsignParser");
+}
+if (typeof PgpSignParser === "undefined") {
+    var PgpSignParser = {};//suppress warnings in editor - doesn't change anything.
+    console.error("pgpparser code loaded before PgpSignParser");
+}
+
+var PgpParsing = {
+    parsers:{
+        CLEARSIGNED: 1,
+        SIGNED: 2
+    },
+    guessMessageType:function(cleartext){
+        //var i1 = cleartext.indexOf("-----BEGIN PGP SIGNED MESSAGE-----");
+        var i2 = cleartext.indexOf("-----BEGIN PGP MESSAGE-----");
```

Comment:
With this change, the clear-signed [MIDI PGP-message](https://pastebin.com/VuyPwuL4) now has an "invalid format", because it contains a PGP signed message (without a clear signature).

To fix this, I think the only solution is to search for `-----BEGIN PGP MESSAGE-----` (or `-----BEGIN PGP SIGNED MESSAGE-----`) at the start of lines only. You couldn't simply search for `-----BEGIN PGP SIGNED MESSAGE-----` (line 39) because even if the signature couldn't possibly contain it, I think theoretically a comment could.

Reply:
That pastebin shows "good signature" for me with the changes. the pastebin contains a space in the inner message headers. 

However you're right that this will produce incorrect results if `-----BEGIN PGP MESSAGE-----` is included in the middle of a string.  guess I will have to use lineparser, splitting, or regex after all.

Reply:
>That pastebin shows "good signature" for me with the changes

Really!? ![The current changes](https://user-images.githubusercontent.com/29614762/160258527-e61ae110-544f-4936-b8e7-c71356fc537d.png)

What about doing instead:
```js
let is_signed = cleartext.startsWith("-----BEGIN PGP MESSAGE-----\n") || cleartext.indexOf("\n-----BEGIN PGP MESSAGE-----\n") !== -1;

return is_signed ? PgpParsing.parsers.SIGNED : PgpParsing.parsers.CLEARSIGNED;
```

Reply:
weird - maybe I have some local changes
![image](https://user-images.githubusercontent.com/3233238/160258923-292537c8-f5fd-45ce-889f-2955487d3d48.png)

well, issue there is that there may be content before the marker lines kick in, and I think I only warn for that.

Reply:
According to the [PGP standard](https://www.rfc-editor.org/rfc/rfc4880#section-7), a valid message can't contain lines starting with dashes. These must be instead escaped by inserting a `- ` at the start of the line (= "dash escaped").
This means that a valid clear signed message will always contain exactly three lines starting with `-----` (`-----BEGIN PGP SIGNED MESSAGE-----`, `-----BEGIN PGP SIGNATURE-----`, `-----END PGP SIGNATURE-----`) and a valid (not clear) signed message will always contain with exactly of these two lines (`-----BEGIN PGP MESSAGE-----`, `-----END PGP MESSAGE-----`).

Thus, it is possible to discriminate between the message type by looking for a `-----BEGIN PGP SIGNATURE-----` line. That's exactly what I do in the code snippet I posted.

Once the message type is determined, the parsers are robust enough to be able to parse as usual the message